frontend/views/components/ContributionItem.vue

```diff
@@ -0,0 +1,160 @@
+<template lang='pug'>
+.c-contribution-item
+  i(:class='iconClass')
+
+  div(v-if='hasWhoElse')
+    transition(name='replacelist')
+      div(
+        v-if='isVisible'
+        key='visible'
+      )
+        .c-contribution-list(v-html='listOfName')
+
+        i18n.is-unstyled.is-link-inherit.link(
+          tag='button'
+          type='button'
+          @click='isVisible = !isVisible'
+        ) Hide
+
+      div(
+        v-else
+        key='hidden'
+        @click='isVisible = !isVisible'
+        v-html='contributionText'
+      )
+
+  .c-contribution-list(
+    v-else=''
+    v-html='contributionText'
+  )
+</template>
+
+<script>
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'ContributionItem',
+  props: {
+    who: [String, Array],
+    what: String,
+    action: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['RECEIVING', 'GIVING'].indexOf(value) !== -1
+      },
+      default: 'RECEIVING'
+    },
+    type: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['NON_MONETARY', 'MONETARY'].indexOf(value) !== -1
+      },
+      default: 'NON_MONETARY'
+    }
+  },
+  data () {
+    return {
+      isVisible: false
+    }
+  },
+  computed: {
+    contributionText () {
+      if (this.hasWhoElse) {
+        const html = {
+          0: `<span class="has-text-bold">${this.what}</span>`,
+          1: `<button class="is-unstyled is-link-inherit link">${this.otherContributor}`,
+          2: '</button>'
+        }
+
+        if (this.action === 'RECEIVING' && this.type === 'MONETARY') {
+          return L('{0} from {1} members{2}', html)
+        } else {
+          return L('A total of {0} to {1} members{2}', html)
+        }
+      } else {
+        if (this.action === 'RECEIVING') {
+          const html = {
+            0: `<span class="has-text-bold">${this.what}</span>`,
+            1: this.firstWho
+          }
+          return L('{0} by {1}', html)
+        } else {
+          if (this.type === 'MONETARY') {
+            const html = {
+              0: `<span class="has-text-bold">${this.what}</span>`,
+              1: this.firstWho
```

Comment:
Same type of changes needed as described [above](https://github.com/okTurtles/group-income-simple/pull/724/files#r342158712) (use descriptive variable names instead of numbers)